domain/consensus/timelock_CLTV_test.go

```diff
@@ -242,9 +329,131 @@ func TestCheckLockTimeVerifyConditionedByAbsoluteTime(t *testing.T) {
+		}
+		defer teardown(false)
+
+		blockAHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{testConsensus.DAGParams().GenesisHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockA: %v", err)
+		}
+		blockBHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockAHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockB: %v", err)
+		}
+		blockCHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockBHash}, nil, nil)
+		if err != nil {
+			t.Fatalf("Error creating blockC: %v", err)
+		}
+		fees := uint64(1)
+		blockDHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockCHash}, nil,
+			[]*externalapi.DomainTransaction{})
+		if err != nil {
+			t.Fatalf("Error creating blockD: %v", err)
+		}
+		blockD, err := testConsensus.GetBlock(blockDHash)
+		if err != nil {
+			t.Fatalf("Failed getting blockD: %v", err)
+		}
+		//Create a CLTV script:
+		timeToWait := int64(12 * 1000)
+		lockTimeTarget := blockD.Header.TimeInMilliseconds() + timeToWait
+		redeemScriptCLTV, err := createScriptCLTV(lockTimeTarget)
+		if err != nil {
+			t.Fatalf("Failed to create a script using createScriptCLTV: %v", err)
+		}
+		p2shScriptCLTV, err := txscript.PayToScriptHashScript(redeemScriptCLTV)
+		if err != nil {
+			t.Fatalf("Failed to create a pay-to-script-hash script : %v", err)
+		}
+		scriptPublicKeyCLTV := externalapi.ScriptPublicKey{
+			Version: constants.MaxScriptPublicKeyVersion,
+			Script:  p2shScriptCLTV,
+		}
+		transactionWithLockedOutput, err := createTransactionWithLockedOutput(blockD.Transactions[transactionhelper.CoinbaseTransactionIndex],
+			fees, &scriptPublicKeyCLTV)
+		if err != nil {
+			t.Fatalf("Error in createTransactionWithLockedOutput: %v", err)
+		}
+		// BlockE contains the locked output (locked by CLTV).
+		// This block should be valid since CLTV script locked only the output.
+		blockEHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockDHash}, nil,
+			[]*externalapi.DomainTransaction{transactionWithLockedOutput})
+		if err != nil {
+			t.Fatalf("Error creating blockE: %v", err)
+		}
+		blockE, err := testConsensus.GetBlock(blockEHash)
+		if err != nil {
+			t.Fatalf("Failed getting blockE: %v", err)
+		}
+		// Create a transaction that tries to spend the locked output.
+		transactionThatSpentTheLockedOutput, err := createTransactionThatSpentTheLockedOutput(transactionWithLockedOutput,
+			fees, redeemScriptCLTV, lockTimeTarget-1)
+		if err != nil {
+			t.Fatalf("Error creating transactionThatSpentTheLockedOutput: %v", err)
+		}
+		emptyCoinbase := externalapi.DomainCoinbaseData{
+			ScriptPublicKey: &externalapi.ScriptPublicKey{
+				Script:  nil,
+				Version: 0,
+			},
+		}
+		var tipHash *externalapi.DomainHash
+		timeStampBlockE := blockE.Header.TimeInMilliseconds()
+		stagingArea := model.NewStagingArea()
+		// Make sure the time limitation has passed.
+		for i := int64(0); ; i++ {
```

Comment:
Why is it needed here and not in `TestCheckLockTimeVerifyConditionedByAbsoluteTime`?

Reply:
Needed in both places.
Already exists there.